### PR TITLE
onshift sharding (proof of concept)

### PR DESCRIPTION
**Pull request by fuhrysteve**

### CODE REVIEW

**Comment by EricWorkman** [comment]
DO NOT MERGE IN

Yes, all caps.

**Comment by fuhrysteve** [comment]
LOL yes DO NOT MERGE this haha

**Comment by mikesnow79** [comment]
Come on, #YOLO

File: turbogears/database.py [hunk 1]
```diff
@@ -98,25 +99,29 @@ def _enable_timestamp_workaround(self, connection):
                 connection.kw["conv"] = conversions
 
         def getConnection(self):
-            try:
-                conn = self.threadingLocal.connection
-                return self.begin(conn)
```
Comment (fuhrysteve) [inline turbogears/database.py]: From @cweisel this needs a little rethinking - it's already *sortof* implementing a "get or create" pattern, so this is a wrapper around a wrapper.. so 2x things to investigate:

1. doing this in threadingLocal
2. thread safe collections?

File: turbogears/database.py [hunk 2]
```diff
@@ -98,25 +99,29 @@ def _enable_timestamp_workaround(self, connection):
...
-            except AttributeError:
-                if self.uri:
-                    conn = sqlobject.connectionForURI(self.uri)
```
Comment (EricWorkman) [inline turbogears/database.py]: This is going to need rework. Up above, in the `__init__`, it accepts a single URI (if none are passed it). We need to change that to accept a URI for each of our shards, and change this piece on ~106 to choose the right URI for the shard.

Comment (fuhrysteve) [inline turbogears/database.py]: Yeah, good point - we'll need to actually *use* the `shard_id` at some point....

**Pull request closed without merging**